### DISCUSSION TITLE
MP4 Video Support?

### DISCUSSION THREAD

**Discussion by Boognishdood** [eduardofilo/RG350_adam_image#233] [Ideas & Requests]
Hi there

I can't seem to find any information on this topic. I'm really new at this, so forgive me if the solution is simple. 

Is there any way to play MP4 files on my RG280v that current has Adam image? I have a bit of extra space on my card and wouldnt mind loading some videos on there if this is possible

**Comment by neilswann80** [comment]
At the moment there's not an OpenDingux Beta (therefore Adam Image) compatible video player.  There is an audio player called [GMU](https://github.com/plrguez/gmu-odbeta/releases/tag/v0.10.2-rc1-odbeta-20211024) that's incorporated into the image.

I'm hoping the video player[ FFPlay](https://github.com/denis-n-kuznetsov/FFmpeg-GCW0) that worked on the original OpenDingux will eventually get recompiled for ODBeta.

**Comment by a-dekker** [comment]
Compiled ffplay from https://github.com/OpenDingux/buildroot

It will be picky, as much depends on the codec, bitrates, resolution of the video and available device screen resolution, limited cpu power etc.
And as a bonus it often hangs when quitting a video :-)

But at least it offers some way of playing videos.
```
X                   pause/play
left/right          seek backward/forward 10 seconds
down/up             seek backward/forward 1 minute
R2/L2               seek backward/forward 10 minutes
select              quit
```

https://github.com/a-dekker/repackaged-opks/raw/main/ffplay.opk

**Reply by neilswann80** [reply]
You legend!

**Comment by felixmreeve** [comment]
@a-dekker this link to the opk seems to be dead? Any chance of a new one?
Thanks!

**Comment by a-dekker** [comment]
Nobody has downloaded it before?
Here's a link that expires in 30 days: https://easyupload.io/0x2o0x

**Reply by neilswann80** [reply]
Thanks muchly @a-dekker!

I've backed it up to the [OneDrive](https://1drv.ms/u/s!AhaEVBCD1ThtiETQygnItfs6Urrp?e=gP1Zf9) games folder.  Any chance you could reupload Nanobounce and Neverball?  Your GitHub hosted files disappeared (any idea as to why?) before I got chance to back these games up to Onedrive.

P.s. it's good to see you around again!  😃

**Comment by felixmreeve** [comment]
Thank you! I have added it to my apps folder and can see it while running Adam, but it seems to try to run and then close/crash? Do I need to have a specific folder to put my media in? Or is there something else going on? Sorry, it doesn't seem to give me much feedback about what's going wrong

**Reply by a-dekker** [reply]
Do not run it using SimpleMenu, but use the default gmenu2x, as it needs to browse for files. By starting it from gmenu2x, you should be presented with a file browser to select your media file to play.

**Reply by felixmreeve** [reply]
ah cool thank you! is there a way to do that with Adam image or would I need to reinstall a different image?

**Reply by a-dekker** [reply]
I do not have Simplemenu as default. In the settings of Simplemenu you can set "Default launcher" to no so it won't start at boot. And perhaps choosing Session-> quit  in settings will also bring you to the default menu, which would be the easiest way.

**Reply by felixmreeve** [reply]
Thank you!! - that's exactly what I was looking for! You're right about it often hanging when quitting and being picky about video format but this is great <3

**Reply by felixmreeve** [reply]
Out of interest - it seems to squash things a bit, is there a format / particular resolution that I can use to avoid this?

**Reply by a-dekker** [reply]
No idea really. The native resolution will be 320x240 or 640x480, and 320x480 for the RG280V. 320x240 would be my best bet.

**Reply by felixmreeve** [reply]
thanks! still seems to squash anything it plays on the rg280v, I'll investigate whether I can output anything that it agrees with

**Reply by a-dekker** [reply]
I do not own this device, but I do frequently read about squashed output in games because of its "non standard" screen. You will find several examples in the game port discussion.
On my 640x480 screen, 320x240 resolution videos do fill the screen without any layout distortion.

**Reply by eduardofilo** [reply]
> ah cool thank you! is there a way to do that with Adam image or would I need to reinstall a different image?

You can create a new system for video adding this to the file `/media/data/local/home/.simplemenu/section_groups/apps and games.ini`:

```
[VIDEO]
execs = /media/data/apps/ffplay.opk
romDirs = /media/sdcard/video/
romExts = .mp4,.avi
```

Add VIDEO to the consoleList parameter at the beginning of the same file.

```
[CONSOLES]
consoleList = GAMES,APPS,VIDEO
```

Then put your video files in a new folder called `video` in the root of the external SD card.
You should now see a new system inside Apps & Games group. The problem will be that there is no art for that system in anyone of the themes. Although you can create manually that art for at least your favorite theme.

**Reply by felixmreeve** [reply]
Ooh this all sounds pretty doable - although I'm actually not sure how to access the files on the INT SD card. Do I need a linux machine?

**Reply by felixmreeve** [reply]
@a-dekker applying a 2x height scale to the video seems to sort out the aspect ratio... bit silly but works fine!

**Reply by felixmreeve** [reply]
@eduardofilo followed your instructions after downloading an iso of ubuntu - works great! Thanks so much, now I just need to figure out how to add an image to the theme!

**Reply by eduardofilo** [reply]
It is best to follow the steps of another system that already has it done. The key is in the .ini file in the root folder of each theme. For example for BigCody, the file is `/media/data/local/home/.simplemenu/themes/640x480/BigCody/theme.ini` (or 320x240 if you have a low resolution console).

At the end of that file you should add something like the following:

```
[VIDEO]
name = Video
logo = resources/video/logo.png
```

And the file `logo.png` with the art should rest in `/media/data/local/home/.simplemenu/themes/640x480/BigCody/resources/video/logo.png`

**Reply by felixmreeve** [reply]
Yes, I got a VIDEOS section working, and it plays videos great, all through simplemenu! thanks very much for your help 😄

**Comment by altharic** [comment]
Is this any good - https://github.com/citral23/FFmpeg

**Reply by a-dekker** [reply]
You can try yourself: https://github.com/citral23/FFmpeg/releases/download/v0.3/FFplay_multi.opk

It's a recent build and should be compiled for OD-beta. Tried one mkv files, but that failed me.
**Edit** tried some other mkv files and they did work.
